MySQL VARCHAR is mapped to `Vec<u8>` instead of `String`

The following is my database schema, I expect the VARCHAR and TEXT types are both mapped to Rust's `String`. However, I got the error in the next following log.
```bash
fyp/slide-service> describe `users`;
+---------------+--------------+------+-----+-------------------+-----------------------------------------------+
| Field         | Type         | Null | Key | Default           | Extra                                         |
+---------------+--------------+------+-----+-------------------+-----------------------------------------------+
| id            | varchar(255) | NO   | PRI | NULL              |                                               |
| username      | varchar(50)  | NO   | UNI | NULL              |                                               |
| email         | varchar(255) | NO   | UNI | NULL              |                                               |
| full_name     | varchar(60)  | YES  |     | NULL              |                                               |
| password_hash | varchar(255) | NO   |     | NULL              |                                               |
| avatar_url    | text         | YES  |     | NULL              |                                               |
| bio           | text         | YES  |     | NULL              |                                               |
| created_at    | timestamp    | NO   |     | CURRENT_TIMESTAMP | DEFAULT_GENERATED                             |
| updated_at    | timestamp    | NO   |     | CURRENT_TIMESTAMP | DEFAULT_GENERATED on update CURRENT_TIMESTAMP |
+---------------+--------------+------+-----+-------------------+-----------------------------------------------+
```

This is the compile-time error
```bash
error[E0308]: mismatched types
  --> cmd/auth-service/src/db/user.rs:23:20
   |
23 |         let user = sqlx::query_as!(User, "SELECT * FROM users WHERE username = ?", username)
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `std::string::String`, found struct `Vec`
   |
   = note: expected struct `std::string::String`
              found struct `Vec<u8>`
   = note: this error originates in the macro `$crate::sqlx_macros::expand_query` (in Nightly builds, run with -Z macro-backtrace for more info)
```

This following is the Rust struct.
```rs
#[derive(Debug, Deserialize, Serialize, sqlx::FromRow, SimpleObject)]
pub struct User {
    pub id: String,
    pub username: String,
    pub email: String,
    #[serde(skip)]
    #[graphql(skip)]
    pub password_hash: String,
    pub full_name: Option<String>,
    pub bio: Option<String>,
    pub avatar_url: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
```

One more note, I'm not sure if this is related, I'm using PlanetScale's MySQL which uses Vitess under the hood.

--- a/sqlx-core/src/mysql/types/str.rs
+++ b/sqlx-core/src/mysql/types/str.rs
@@ -12,6 +12,7 @@ const COLLATE_UTF8_UNICODE_CI: u16 = 192;
 const COLLATE_UTF8MB4_UNICODE_CI: u16 = 224;
 const COLLATE_UTF8MB4_BIN: u16 = 46;
 const COLLATE_UTF8MB4_GENERAL_CI: u16 = 45;
+const COLLATE_UTF8MB4_0900_AI_CI: u16 = 255;
 
 impl Type<MySql> for str {
     fn type_info() -> MySqlTypeInfo {
@@ -42,6 +43,7 @@ impl Type<MySql> for str {
                 | COLLATE_UTF8_GENERAL_CI
                 | COLLATE_UTF8MB4_BIN
                 | COLLATE_UTF8MB4_GENERAL_CI
+                | COLLATE_UTF8MB4_0900_AI_CI
         )
     }
 }

